### DISCUSSION TITLE
Error when destroying StackedBar

### DISCUSSION THREAD

**Discussion by errantmind** [f5/unovis#206] [Q&A] [answered]
I'm using Vue, which doesn't seem to have its own package so instead I'm writing vanilla javascript to create and update charts.

Upon the data being updated, I first try to remove and destroy the previously rendered chart before rendering again. I couldn't find a clear explanation on how to do this in the docs so I'm calling destroy directly. This is the flow:

```
const container = document.querySelector('#myplot');
const legend = document.querySelector('#mylegend');

// Remove any previous charts

if (this.chartReference) {
   this.chartReference.destroy();
   this.chartReference = undefined;
}

while (container.firstChild) {
   container.removeChild(container.firstChild);
}

if (this.legendReference) {
   this.legendReference.destroy();
   this.legendReference = undefined;
}

while (legend.firstChild) {
   legend.removeChild(legend.firstChild);
}

// Create the new chart

this.legendReference = new BulletLegend(legend, {
    <redacted>
 });

const bar = new StackedBar({
    <redacted>
 });

this.chartReference = new XYContainer(container, {
  components: [ bar ],
  height: this.chartData.length * 20,
  xAxis: new Axis({
     <redacted>
  }),
  yAxis: new Axis({
     tickTextWidth: this.isSmallscreen ? 75 : null,
     tickTextFitMode: FitMode.Trim,
     tickFormat: (_, i) => {
       return this.chartData[i].x;      
     },
     label: this.isSmallscreen ? null : 'Site',
     numTicks: this.chartData.length,
  }),
}, this.chartData);
```

The initial render on the page being first mounted is fine, but i get an error on re-renders:

```
caught TypeError: Cannot read properties of undefined (reading 'observationTargets')
    at ResizeObserverController2.disconnect (ResizeObserverController.js:44:16)
    at ResizeObserver2.disconnect (ResizeObserver.js:32:34)
    at Proxy.destroy (index.ts:136:27)
    at Proxy.destroy (index.ts:399:11)
    at Proxy.drawPlot2 (SummaryPage.vue?t=1684771471298:121:36)
    at Proxy.updateState (SummaryPage.vue?t=1684771471298:256:18)
    at Proxy.applyFilter (SummaryPage.vue?t=1684771471298:302:15)
    at callWithErrorHandling (runtime-core.esm-bundler.js:173:22)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js:182:21)
    at emit (runtime-core.esm-bundler.js:730:9)
```

The last user controlled callsite in the stack trace is the line above where I call `this.chartReference.destroy();`

Am I doing something wrong?

**Comment by rokotyan** [comment, marked as answer]
@errantmind Your code looks correct, so it's hard to tell what's going wrong. If you could share an example on StackBlitz or somewhere else, we'll happy to take a look.

**Reply by errantmind** [reply]
Sure, is this the right way to go about re-rendering a chart? Or is there some other approach? Here is the StackBlitz for a minimal example:

https://stackblitz.com/edit/vue-4nca51?file=src%2FApp.vue

If you comment out the 're-rendering' in `mounted` it renders.

**Reply by rokotyan** [reply]
@errantmind Thanks, I just had a quick look and the reason is not clear to me. It looks like a Vue specific issue, as I couldn't reproduce it in our dev environment. 

I'll take another look later, and if you just need to update the chart with new data or config there's no reason to destroy it, you can simply use the `setData(newData)` and `setConfig(newConfig)` methods.

**Reply by errantmind** [reply]
Thanks, I'll try that.

**Reply by errantmind** [reply]
That worked, although in my case I had to use `updateContainer` and not `setConfig`.

**Reply by rokotyan** [reply]
@errantmind Great! You're right, the method for updating the container is called `updateContainer`, not `setConfig`.
